realm/realm-library/src/androidTest/java/io/realm/RealmChangeListenerTests.java

```diff
@@ -0,0 +1,217 @@
+import io.realm.rule.RunInLooperThread;
+import io.realm.rule.RunTestInLooperThread;
+import io.realm.rule.TestRealmConfigurationFactory;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+
+@RunWith(AndroidJUnit4.class)
+public class RealmChangeListenerTests {
+    @Rule
+    public final UiThreadTestRule uiThreadTestRule = new UiThreadTestRule();
+    @Rule
+    public final RunInLooperThread looperThread = new RunInLooperThread();
+    @Rule
+    public final TestRealmConfigurationFactory configFactory = new TestRealmConfigurationFactory();
+
+    private Realm realm;
+    private RealmConfiguration realmConfig;
+
+    @Before
+    public void setUp() {
+        realmConfig = configFactory.createConfiguration();
+    }
+
+    @After
+    public void tearDown() {
+        if (realm != null) {
+            realm.close();
+        }
+    }
+
+    @Test
+    @RunTestInLooperThread
+    public void returnedRealmIsNotNull() {
+        looperThread.realm.addChangeListener(new RealmChangeListener<Realm>() {
+            @Override
+            public void onChange(Realm realm) {
+                assertNotNull(realm);
+                assertFalse(realm.isClosed());
+                looperThread.testComplete();
+            }
+        });
+        looperThread.realm.beginTransaction();
+        looperThread.realm.commitTransaction();
+    }
+
+    @Test
+    @RunTestInLooperThread
+    public void returnedDynamicRealmIsNotNull() {
+        final DynamicRealm dynamicRealm = DynamicRealm.getInstance(looperThread.realmConfiguration);
+        dynamicRealm.addChangeListener(new RealmChangeListener<DynamicRealm>() {
+            @Override
+            public void onChange(DynamicRealm dynRealm) {
+                assertNotNull(dynRealm);
+                assertFalse(dynRealm.isClosed());
+                dynRealm.close();
+                looperThread.testComplete();
+            }
+        });
+        dynamicRealm.beginTransaction();
+        dynamicRealm.commitTransaction();
+    }
+
+    @Test
+    @RunTestInLooperThread
+    public void returnedRealmResultsIsNotNull() {
+        RealmResults<Cat> cats = looperThread.realm.where(Cat.class).findAll();
+        cats.addChangeListener(new RealmChangeListener<RealmResults<Cat>>() {
+            @Override
+            public void onChange(RealmResults<Cat> result) {
+                assertEquals("cat1", result.first().getName());
+                looperThread.testComplete();
+            }
+        });
+        looperThread.realm.beginTransaction();
+        Cat cat = looperThread.realm.createObject(Cat.class);
+        cat.setName("cat1");
+        looperThread.realm.commitTransaction();
+    }
+
+    @Test
+    @RunTestInLooperThread
+    public void returnedRealmResultsOfModelIsNotNull() {
+        RealmResults<AllTypesRealmModel> alltypes = looperThread.realm.where(AllTypesRealmModel.class).findAll();
```

Comment:
Perhaps refactor `looperThread.realm` into its own variable:

```
Realm realm = looperThread.realm;
...
```

It makes the tests easier to read.